frontend/src/mllike.js

```diff
@@ -75,28 +115,44 @@
                 stream.skipToEnd();
                 return 'comment';
             }
+
+            /* match digits */
             if (/\d/.test(ch)) {
                 stream.eatWhile(/[\d]/);
+                /* match floating numbers */
                 if (stream.eat('.')) {
                     stream.eatWhile(/[\d]/);
                 }
+
+                state.indentHint = INDENT_OPTIONS.STAY;
+
                 return 'number';
             }
             if ( /[+\-*&%=<>!?|]/.test(ch)) {
                 return 'operator';
             }
+
             if (/[\w\xa1-\uffff]/.test(ch)) {
                 stream.eatWhile(/[\w\xa1-\uffff]/);
                 var cur = stream.current();
                 if (words.hasOwnProperty(cur)) {
-                    if (stream.eol() && autoIndent && (cur == 'let' || cur == 'in' ||cur == 'local' || cur == 'struct' || cur == 'sig')) {
-                        state.indentHint += 2;
+
+                    var matchedObject = words[cur];
+                    var shouldIndent = matchedObject.hasOwnProperty('indent')
+                        && matchedObject.indent;
+
+                    if (stream.eol() && shouldIndent) {
+                        state.indentHint = INDENT_OPTIONS.INDENT;
                     }
-                    return words[cur];
+                    return matchedObject.type;
                 } else {
                     return 'variable';
```

Comment:
For example setting `state.indentHint = INDENT_OPTIONS.STAY` here will already fix a lot of corner cases.

Reply:
This sounds good, I added it in line 183 in 879c128f01bd39ad1c67e9d60d515f423df57ad3